scripts/run_one_ks.sh

```diff
@@ -111,37 +111,47 @@ runone() {
     cp ${tmpdir}/virt-install.log ${tmpdir}/virt-install-human.log
     sed -i 's/#012/\n/g' ${tmpdir}/virt-install-human.log
     echo
-    if [[ -f ${tmpdir}/virt-install.log && "$(grep CRIT ${tmpdir}/virt-install.log)" != "" ]]; then
-        echo "RESULT:${name}:FAILED:\n$(grep CRIT ${tmpdir}/virt-install.log | sed 's/#012/\n/g')"
-        cleanup ${tmpdir}
-        cleanup_tmp ${tmpdir}
-        return 1
-    elif [[ -f ${tmpdir}/livemedia.log ]]; then
-        if [[ "$(grep 'due to timeout' ${tmpdir}/livemedia.log)" != "" ]]; then
-            echo "RESULT:${name}:FAILED:Test timed out."
-            cleanup ${tmpdir}
-            cleanup_tmp ${tmpdir}
-            return 2
-        elif [[ "$(grep 'Call Trace' ${tmpdir}/livemedia.log)" != "" ]]; then
-            echo "RESULT:${name}:FAILED:Kernel panic."
-            cleanup ${tmpdir}
-            cleanup_tmp ${tmpdir}
-            return 0
+
+    RESULT=""
+    RET_CODE=0
+    if [ -f ${tmpdir}/virt-install.log ]; then
+        # Ignore rsyslogd CRIT error, this doesn't block the installation
+        if [[ "$(grep -e 'CRIT systemd-coredump:[^\n]*rsyslogd' ${tmpdir}/virt-install.log)" != "" ]]; then
+            echo "CRIT error in test:$(grep CRIT ${tmpdir}/virt-install.log \
+                  | sed 's/#012/\n/g')"
+        # Anaconda CRIT error blocking the installation
+        elif [[ "$(grep CRIT ${tmpdir}/virt-install.log)" != "" ]]; then
+            RESULT="FAILED:$(grep CRIT ${tmpdir}/virt-install.log | sed 's/#012/\n/g')"
+            RET_CODE=1
         fi
 
-        result=$(validate ${tmpdir})
-        if [[ $? != 0 ]]; then
-            echo "RESULT:${name}:FAILED:${result}"
-            cleanup ${tmpdir}
-            cleanup_tmp ${tmpdir}
-            return 1
+        if [[ ${RET_CODE} -eq 0 ]]; then
+            # TIME OUT error
+            if [[ "$(grep 'due to timeout' ${tmpdir}/livemedia.log)" != "" ]]; then
+                RESULT="FAILED:Test timed out."
+                RET_CODE=2
+            # Kernel Call Trace error
+            elif [[ "$(grep 'Call Trace' ${tmpdir}/livemedia.log)" != "" ]]; then
+                RESULT="FAILED:Kernel panic."
+                RET_CODE=0
+            fi
         fi
     fi
 
-    echo RESULT:${name}:SUCCESS
+    ret=$(validate ${tmpdir})
+    if [[ $? != 0 ]]; then
+        RESULT="FAILED:${ret}"
+        RET_CODE=1
+    fi
+
+    if [ -z "${RESULT}" ]; then
+        RESULT="SUCCESS"
+    fi
```

Comment:
This all looks good to me, but I don't like the mixture of ``[`` and ``[[``. Please use only one or another (preferably ``[[``), but not both.